src/main/java/dev/blachut/svelte/lang/codeInsight/SvelteReferencesSearch.kt

```diff
@@ -26,6 +28,24 @@ class SvelteReferencesSearch : QueryExecutorBase<PsiReference, ReferencesSearch.
         val containingFile = element.containingFile
         val identifier = (element as? JSElement)?.name ?: return
 
+        if (element is JSVariable) {
+            val usageText = "\$$identifier"
+
+            val manager = PsiManager.getInstance(element.project)
+            val files = FileTypeIndex.getFiles(SvelteHtmlFileType.INSTANCE, GlobalSearchScope.allScope(element.project))
+                .mapNotNull { manager.findFile(it) }
+            val scope = LocalSearchScope(files.toTypedArray())
```

Comment:
I've tired with 
```
GlobalSearchScope.filesScope(
  element.project,
  FileTypeIndex.getFiles(SvelteHtmlFileType.INSTANCE, GlobalSearchScope.allScope(element.project))
)
```
and
```
GlobalSearchScope.getScopeRestrictedByFileTypes(GlobalSearchScope.allScope(element.project), SvelteHtmlFileType.INSTANCE)
```
but they are not working.

I wonder if this is performant

Reply:
@tomblachut Hi!

Could please share a code sample when other approaches don't work? I've tried similar code and in simple cases, everything works just fine. 
```
val scope = if (effectiveSearchScope is GlobalSearchScope)
                GlobalSearchScope.getScopeRestrictedByFileTypes(effectiveSearchScope, SvelteHtmlFileType.INSTANCE)
            else effectiveSearchScope
  
queryParameters.optimizer.searchWord(usageText, scope, 
  UsageSearchContext.IN_CODE, true, element, SubstituteReferenceProcessor(element, usageText))
```

There are several issues with the current implemented solution 
```
....
LocalSearchScope(files.toTypedArray())
```

1. It doesn't respect ``queryParameters`` scope. As a result, it will start a search over the whole project even if we need to highlight usages only in the current file (in this case ``queryParameters.effectiveSearchScope`` usually is ``LocalSearchScope(currentFile)``)
2. It processes all files every time and I believe it can be very slow for big svelte projects.

Reply:
I just assigned previous snippet to `val scope`, rest of the class is the same. 

Your code is also not working for me, example find usages from screenshot above shows only line with import, weird.

Reply:
>Your code is also not working for me, example find usages from screenshot above shows only line with import, weird.

oh, yes, it doesn't work for "find usages" but works for local searches e.g. highlightings. I am searching for a better solution.

Reply:
thx!

Reply:
I see. You don't have an id indexer implementation so reference searcher cannot find identifiers inside svelte interpolations (and it works for script tags because of ``com.intellij.psi.impl.cache.impl.idCache.HtmlIdIndexer``)

You need something like this: [VueIdIndexer](https://github.com/JetBrains/intellij-plugins/blob/master/vuejs/src/org/jetbrains/vuejs/lang/html/index/VueIdIndexer.kt) and it will solve the magic problem with scopes and also #145 (I still believe it should work without additional actions if all search implementations are correct).

Reply:
just a note: it worked for ``LocalSearchScope`` because it uses a different algorithm of searching. While ``GlobalSearchScope`` uses indexes  ``LocalSearchScope`` just scan the provided scope

Reply:
I'll take a look at that :+1:

Reply:
Awesome, now Find Usages working with `GlobalSearchScope`, and it also fixed some weird highlighting offset bugs that I had.

#145 is still broken, unfortunately

Reply:
hm, I will take a look again today